sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnection.java

```diff
@@ -52,6 +55,8 @@
     private final TransactionType transactionType;
     
     private final ShardingTransactionManager shardingTransactionManager;
+    private transient Object transaction = null;
+    private transient Boolean originAutoCommit = null;
```

Comment:
ShardingConnection is not pooledConnection, it will new from datasource everytime, so we should not keep transaction and originalAutoCommit status here.